lib/curses_renderer.rb

```diff
@@ -1,103 +1,111 @@
 # encoding: UTF-8
 
-require 'curses'                # Require the Curses gem so we can play with terminal output and extend input options for our rendering class. 
+require 'curses'
 
-class CursesRenderer            # Create a constant for arrow key movement. 
-  MOVEMENT_KEYS = [Curses::Key::UP, Curses::Key::DOWN, Curses::Key::LEFT, Curses::Key::RIGHT].freeze            
+class CursesRenderer
+  MOVEMENT_KEYS = [Curses::Key::UP, Curses::Key::DOWN, Curses::Key::LEFT, Curses::Key::RIGHT].freeze
+  PAINT_KEYS = [Curses::Key::ENTER, ' ', "\n"]
+  UNPAINTED = 1
+  Curses.init_pair(UNPAINTED, Curses::COLOR_WHITE, Curses::COLOR_BLACK)
+  PAINTED = 2
+  Curses.init_pair(PAINTED, Curses::COLOR_WHITE, Curses::COLOR_RED)
```

Comment:
L9/L11 — these are imperative calls into Curses, but where they're at, in the class body, they will be executed when this class loads. Depending on when it loads, it may work or not, but I don't think so. Color is turned on within the `initialize_curses` method, and these will be run _before_ that, since the class is loaded before the method can be called.

Thus the `init_pair` methods should probably be considered part of the curses initialization step, and handled in the method we have for that already.